src/Components/Table/types.ts

```diff
@@ -0,0 +1,24 @@
+import { SxProps } from "@mui/material";
+import { MouseEvent } from "react";
+import { GridRenderCellParams, GridSortModel } from "@mui/x-data-grid";
+import { sortType } from "./consts";
+export type TableCol = {
+  title: string;
+  field: string;
+  width?: number;
+  flex?: number;
+  minWidth?: number;
+  renderCell?: (params: GridRenderCellParams) => React.ReactNode;
+};
+
+export type TableProps = {
+  cols: TableCol[];
+  rows: any[];
+  onRowClick: (event: MouseEvent<HTMLElement>) => void;
```

Comment:
pay attention that onRowClick is not an array, but a single function.
Meaning that each row will have the same click callback. This is not right or wrong , just an implementation decision that you indirectly took.

Reply:
For our application, we need the same function for all rows, because clicking on a row in the table should show the card in modal.